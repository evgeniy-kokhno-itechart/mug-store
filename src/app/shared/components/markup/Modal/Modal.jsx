import React from 'react';
import { PropTypes } from 'prop-types';
import './Modal.scss';

const Modal = ({
  title, text, textConfirm, textAbort, closeBtnClasses, confirmBtnClasses, onClose, onConfirm,
}) => (
  <div className="modal-page">
    <div className="modal-page__content">
      <h1 className="modal-page__header">
        {title}
      </h1>
      <div className="modal-page__body">
        {text}
      </div>
      <div className="modal-page__buttons">
        <button type="button" className={`button button--solid ${closeBtnClasses}`} onClick={onClose}>
          {textAbort}
        </button>
        <button type="button" className={`button button--solid ${confirmBtnClasses}`} onClick={onConfirm}>
          {textConfirm}
        </button>
      </div>
    </div>
  </div>
);

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  textConfirm: PropTypes.string.isRequired,
  textAbort: PropTypes.string.isRequired,
  closeBtnClasses: PropTypes.string,
  confirmBtnClasses: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

Modal.defaultProps = {
  closeBtnClasses: '',
  confirmBtnClasses: '',
};

export default Modal;
