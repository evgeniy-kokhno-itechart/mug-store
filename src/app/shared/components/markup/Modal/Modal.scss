@import "Variables";

.modal-page {
    position: fixed;
    height: 100vh;
    width: 100%;
    background-color: rgba($color-primary-dark, 0.8);
    left: 0;
    top: 0;
    z-index: 9999;

    &__content {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        background-color: $color-white;
        box-shadow: 0 2rem 4rem rgba($color-black, 0.2);
        padding: 2rem;
        animation: openModal 0.5s ease-in-out;
    }
    &__header {
        font-size: 2rem;
        text-align: center;

    }

    &__body {
        text-align: center;
        padding: 2rem 0.5rem;
        border-bottom: 1px solid $color-primary-dark;
    }

    &__buttons {
        display: flex;
        justify-content: space-between;
        margin: 0.5rem 1rem 0;
    }

    @keyframes openModal {
        0% {
            opacity: 0;
            transform: translate(-50%, -50%) scale(0.5);
        }

        80% {
            transform: translate(-50%, -50%) scale(1.05);
        }
    
        100% {
            opacity: 1;
            transform: translate(-50%, -50%) scale(1);
        }
    }
     
}