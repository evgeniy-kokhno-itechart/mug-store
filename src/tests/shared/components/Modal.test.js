/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Modal from '../../../app/shared/components/markup/Modal';

configure({ adapter: new Adapter() });

describe('<Modal />', () => {
  let ModalWrapper;

  beforeEach(() => {
    const onConfirm = jest.fn();
    ModalWrapper = shallow(
      <Modal
        id="100"
        buttonLabel="Test Button"
        buttonClasses="testbuttonclasses"
        title="Test Title"
        text="Test Text"
        textConfirm="Test confirmation text"
        textAbort="Test Abort"
        onConfirm={onConfirm}
      />,
    );
  });

  it('check Modal renders properly with default props', () => {
    expect(ModalWrapper).toMatchSnapshot();
  });
});
