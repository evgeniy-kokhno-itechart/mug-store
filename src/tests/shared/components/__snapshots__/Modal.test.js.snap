// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> check Modal renders properly with default props 1`] = `
<div
  className="modal-page"
>
  <div
    className="modal-page__content"
  >
    <h1
      className="modal-page__header"
    >
      Test Title
    </h1>
    <div
      className="modal-page__body"
    >
      Test Text
    </div>
    <div
      className="modal-page__buttons"
    >
      <button
        className="button button--solid "
        onClick={[MockFunction]}
        type="button"
      >
        Test Abort
      </button>
      <button
        className="button button--solid "
        onClick={[MockFunction]}
        type="button"
      >
        Test confirmation text
      </button>
    </div>
  </div>
</div>
`;
